arch/arm64/src/common/arm64_internal.h

```diff
@@ -0,0 +1,387 @@
+ * value aligned the 8-bytes as required by the ARM EABI.
+ */
+
+#define SMP_STACK_SIZE       STACK_ALIGN_UP(CONFIG_IDLETHREAD_STACKSIZE)
+
+THREAD_STACK_ARRAY_DEFINE_EXTERN(g_cpu_idlestackalloc, CONFIG_SMP_NCPUS,
+                          SMP_STACK_SIZE);
+THREAD_STACK_ARRAY_DEFINE_EXTERN(g_interrupt_stacks, CONFIG_SMP_NCPUS,
+                          INTSTACK_SIZE);
+uintptr_t arm64_intstack_alloc(void);
+uintptr_t arm64_intstack_top(void);
+#else
+/* idle thread stack for primary core */
+
+THREAD_STACK_DEFINE_EXTERN(g_idle_stack, CONFIG_IDLETHREAD_STACKSIZE);
+THREAD_STACK_DEFINE_EXTERN(g_interrupt_stack, INTSTACK_SIZE);
+#endif
+
+/* This is the beginning of heap as provided from arm64_head.S.
+ * This is the first address in DRAM after the loaded
+ * program + bss + idle stack.  The end of the heap is
+ * CONFIG_RAM_END
+ */
+
+/* Address of the saved user stack pointer */
+
+#if CONFIG_ARCH_INTERRUPTSTACK > 3
+EXTERN uint64_t g_intstackalloc; /* Allocated stack base */
+EXTERN uint64_t g_intstacktop;   /* Initial top of interrupt stack */
+#else
+#  error CONFIG_ARCH_INTERRUPTSTACK must be defined (4096 at least) at arm64
+#endif
+
+/* These 'addresses' of these values are setup by the linker script.  They
+ * are not actual uint64_t storage locations! They are only used
+ * meaningfully in the following way:
+ *
+ *  - The linker script defines, for example, the symbol_sdata.
+ *  - The declaration extern uint64_t _sdata; makes C happy.  C will believe
+ *    that the value _sdata is the address of a uint64_t variable _data
+ *    (it is not!).
+ *  - We can recover the linker value then by simply taking the address of
+ *    of _data.  like:  uint64_t *pdata = &_sdata;
+ *
+ * Memory layout for Nuttx at arm64 for FLAT Build
+ *
+ *    +-------------------------+ <- RAM BASE
+ *    |                         |
+ *    |<<<<<<<<<<<<<<<<<<<<<<<<<| <- LOAD_BASE
+ *    |   text(code) section    |
+ *    |   vector table          |
+ *    +-------------------------+-- page align(4096)
+ *    |   rodata section        |
+ *    +-------------------------+-- page align(4096)
+ *    |   data/bss section      |
+ *    +-------------------------+-- page align(4096)
+ *    |   initstack section     |
+ *    +-------------------------+-- page align(4096)
+ *    |   heap alloc area       |
+ *    |                         |
+ *    |                         |
+ *    |                         |
+ *    +-------------------------+
+ *
+ * Note:
+ *  1. initstack is for idle_thread and interrupt stack,
+ *    it has dedicated stack for IRQ handler in arm64
+ *  2. Every section with different memory attribute,
+ *    please check mmu_nxrt_regions at arm64_mmu.c
+ *
+ * please check dramboot.ld at specified platform for more detail
+ */
+
+EXTERN char _stext[];            /* Start of .text */
+EXTERN char _etext[];            /* End of .text */
+EXTERN char _sztext[];           /* Size of .text */
```

Comment:
let's remove all _szxxx? since we can compute them from _exxxx - _sxxxx

Reply:
the size is used at the struct mmu_nxrt_regions of arm64_mmu.c 
the structure is defined as const and put it into rodata section.
I want set the rodata as RO attribute to avoid illegal access to the section. it will be done when I used the MMU/MPU completely
but if i use _exxxx - _sxxxx, it will trigger compile error